examples/mujoco/mujoco_env.py

```diff
@@ -68,14 +71,24 @@ def __init__(self, task: str, seed: int, obs_norm=True) -> None:
         )
         self.obs_norm = obs_norm
 
+    def create_venv(self, num_envs: int, mode: EnvMode) -> BaseVectorEnv:
+        """Create vectorized environments.
+
+        :param num_envs: the number of environments
+        :param mode: the mode for which to create
+        :return: the vectorized environments
+        """
+        env = super().create_venv(num_envs, mode)
+        if self.obs_norm:
+            env = VectorEnvNormObs(env, update_obs_rms=mode == EnvMode.TRAIN)
+        return env
+
     def create_envs(self, num_training_envs: int, num_test_envs: int) -> ContinuousEnvironments:
         envs = super().create_envs(num_training_envs, num_test_envs)
         assert isinstance(envs, ContinuousEnvironments)
 
         # obs norm wrapper
         if self.obs_norm:
-            envs.train_envs = VectorEnvNormObs(envs.train_envs)
-            envs.test_envs = VectorEnvNormObs(envs.test_envs, update_obs_rms=False)
             envs.test_envs.set_obs_rms(envs.train_envs.get_obs_rms())
```

Comment:
Do you think we need this line? I believe it might have been obsolete in the original implementation as well.

Reply:
@MischaPanch the idea behind this line is related to the small discussion we had recently about whether to handle this sort of thing in the env or the agent. The trouble is, if you have learnt stuff in the env, you literally have to transfer all of it from the training envs to the test envs before every test step, which is obviously a PITA. If it's associated with the agent, there is far less that can go wrong.

Reply:
To make the change backwards compatible, I'd suggest that we change the interface by adding `create_watch_env: bool = True`.

Reply:
Right, I missed the part where we would like to watch during train. I guess then there is no other way than adding the watch_env during `create_envs` (with everything what follows). Always adding it is not a good idea, I think. At least for me, this opens a mujoco viewer upon env creation.

Reply:
Yes, watching should be optional.. I'm generally not happy about the current watch Dynamics (users shouldn't be forced or entonces to use tianshou abstractions to watch a trained policy), but that's a story for another PR. For now, let's just make it optional and easy to turn off from any interface

Reply:
@MischaPanch, watching has always been optional, and users aren't forced to use Tianshou abstractions for it either; they just have an option, which, however, as a post-training stage, doesn't really make sense to use for learning processes that take more than a few minutes. But even for longer learning processes, users can still conveniently use the watch function by reloading the experiment as follows:

```
    e = Experiment.from_directory("wherever")
    e.config.watch = True
    e.config.train = False
    e.run()
```

This change is purely about when/where the watch env is created and making it receive the obs_rms from the train env; and it looks like we'll have to create it early if we are to gracefully handle the sharing of the obs rms sharing - a problem which could be avoided entirely if the normalisation was handled in the agent rather than the environment; that's why I mentioned you.

Reply:
In case it wasn't clear, I never suggested that we should create the watch_env by default. I suggested the opposite.

Reply:
I misunderstood the original reason you pinged me.

As mentioned in a private conversation, binding this kind of processing to the agent has mostly upsides, but also some downsides. Let's have a chat about it next week. It's not within the scope of this PR, right?